collector/hyperv.go

```diff
@@ -496,6 +512,84 @@ func NewHyperVCollector() (Collector, error) {
 			[]string{"adapter"},
 			nil,
 		),
+
+		//
+
+		VMStorageErrorCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "error_count"),
+			"This counter represents the total number of errors that have occurred on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+		VMStorageQueueLength: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "queue_length"),
+			"This counter represents the current queue length on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+		VMStorageReadBytesPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "bytes_read"),
+			"This counter represents the total number of bytes that have been read per second on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+		VMStorageReadOperationsPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "operations_read"),
+			"This counter represents the number of read operations that have occurred per second on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+		VMStorageWriteBytesPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "bytes_written"),
+			"This counter represents the total number of bytes that have been written per second on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+		VMStorageWriteOperationsPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_storage"), "operations_written"),
+			"This counter represents the number of write operations that have occurred per second on this virtual device",
+			[]string{"vm_storage"},
+			nil,
+		),
+
+		//
+
+		VMNetworkBytesReceivedPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, buildSubsystemName("vm_network"), "bytes_received"),
+			"This counter represents the total number of bytes received per second by the network adapter",
+			[]string{"vm_network"},
```

Comment:
As above, perhaps in this case `adapter` or `interface`?

Reply:
I went for *interface* here